Updating argocd_cluster fails with 5.5

<!---
Hi there,

Thank you for opening an issue. Please provide as much detail as possible when reporting issue as this benefits all parties and will most likely lead to a quicker resolution of the issue.
--->


### Terraform Version, ArgoCD Provider Version and ArgoCD Version
<!--- Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed. --->
```
Terraform version: 1.4.6
ArgoCD provider version: 5.5
ArgoCD version: 2.7.4
```

### Affected Resource(s)
<!-- Please list the resources as a list, for example:
- argocd_cluster
If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this. -->

- argocd_cluster

### Terraform Configuration Files
```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

```hcl
resource "argocd_cluster" "test" {
  name = "..."
  config {
    exec_provider_config {
       args = [...]
    }
  }
}
```

### Panic Output
<!--If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log` -->

```
Stack trace from the terraform-provider-argocd_v5.5.0 plugin:

panic: interface conversion: interface {} is []interface {}, not []string

goroutine 101 [running]:
github.com/oboukili/terraform-provider-argocd/argocd.flattenClusterConfigExecProviderConfig(0xc001115040, 0xc001022870?)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_cluster.go:242 +0x2eb
github.com/oboukili/terraform-provider-argocd/argocd.flattenClusterConfig({{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, {0x0, 0x0}, {0x0, ...}, ...}, ...}, ...)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_cluster.go:180 +0xbc
github.com/oboukili/terraform-provider-argocd/argocd.flattenCluster(0xc0013b8b40, 0x34f6400?)
	github.com/oboukili/terraform-provider-argocd/argocd/structure_cluster.go:139 +0x325
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDClusterRead({0x34f6400, 0xc0012f5080}, 0xc0011d5180, {0x2c43120?, 0xc000f83c70})
	github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_cluster.go:108 +0x270
github.com/oboukili/terraform-provider-argocd/argocd.resourceArgoCDClusterUpdate({0x34f6400, 0xc0012f5080}, 0xc0011d5180, {0x2c43120?, 0xc000f83c70?})
	github.com/oboukili/terraform-provider-argocd/argocd/resource_argocd_cluster.go:134 +0x445
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc000d6f340, {0x34f6438, 0xc0011c3320}, 0xd?, {0x2c43120, 0xc000f83c70})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:741 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000d6f340, {0x34f6438, 0xc0011c3320}, 0xc000a0a680, 0xc0011d4d00, {0x2c43120, 0xc000f83c70})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:847 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0009950e0, {0x34f6438?, 0xc0011c3200?}, 0xc00066b310)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00097da40, {0x34f6438?, 0xc0011c29f0?}, 0xc000670700)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x2d789a0?, 0xc00097da40}, {0x34f6438, 0xc0011c29f0}, 0xc000670690, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004be000, {0x3503180, 0xc0009c6680}, 0xc0011c0fc0, 0xc000d8c990, 0x4d4f260, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0004be000, {0x3503180, 0xc0009c6680}, 0xc0011c0fc0, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-argocd_v5.5.0 plugin crashed!
```

### Steps to Reproduce
<!-- Please list the steps required to reproduce the issue, for example:
1. `terraform apply` -->

Existing cluster with `exec_provider_config.args` defined previously always suggested change. After 5.5 it still suggested change, but now marking the property as sensitive. When applying it then failed with this error message.

### Expected Behavior
<!-- What should have happened? -->

Either:

1. Suggesting change after upgrading to 5.5 and running successfully first time and after that not suggesting change again.
2. Not suggest change at all after upgrade to 5.5

### Actual Behavior
<!-- What actually happened? -->

Crashing on apply

### Important Factoids
<!-- Are there anything atypical about your accounts that we should know? For example: For general provider/connectivity issues, how is your ArgoCD server exposed? When dealing with cluster resources, what type of cluster are you referencing (where is it hosted)?  -->

### References
<!--Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- GH-1234
-->

### Community Note
<!--- Please keep this note for the community --->
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

--- a/argocd/structure_cluster.go
+++ b/argocd/structure_cluster.go
@@ -195,11 +195,11 @@ func flattenClusterConfig(config application.ClusterConfig, d *schema.ResourceDa
 	// attributes and load them from state instead.
 	// See https://github.com/argoproj/argo-cd/blob/8840929187f4dd7b9d9fd908ea5085a006895507/server/cluster/cluster.go#L448-L466
 	if bt, ok := d.GetOk("config.0.bearer_token"); ok {
-		r["bearer_token"] = bt.(string)
+		r["bearer_token"] = bt
 	}
 
 	if p, ok := d.GetOk("config.0.password"); ok {
-		r["password"] = p.(string)
+		r["password"] = p
 	}
 
 	return []map[string]interface{}{r}
@@ -217,7 +217,7 @@ func flattenClusterConfigTLSClientConfig(tcc application.TLSClientConfig, d *sch
 	// the state of this attribute and load it from state instead.
 	// See https://github.com/argoproj/argo-cd/blob/8840929187f4dd7b9d9fd908ea5085a006895507/server/cluster/cluster.go#L448-L466
 	if kd, ok := d.GetOk("config.0.tls_client_config.0.key_data"); ok {
-		c["key_data"] = kd.(string)
+		c["key_data"] = kd
 	}
 
 	return []map[string]interface{}{c}
@@ -238,12 +238,12 @@ func flattenClusterConfigExecProviderConfig(epc *application.ExecProviderConfig,
 	// sensitive data. Thus, we can't track the state of these
 	// attributes and load them from state instead.
 	// See https://github.com/argoproj/argo-cd/blob/8840929187f4dd7b9d9fd908ea5085a006895507/server/cluster/cluster.go#L454-L461
-	if a, ok := d.GetOk("config.0.exec_provider_config.0.args"); ok {
-		c["args"] = a.([]string)
+	if args, ok := d.GetOk("config.0.exec_provider_config.0.args"); ok {
+		c["args"] = args
 	}
 
-	if a, ok := d.GetOk("config.0.exec_provider_config.0.env"); ok {
-		c["env"] = a.([]string)
+	if env, ok := d.GetOk("config.0.exec_provider_config.0.env"); ok {
+		c["env"] = env
 	}
 
 	return []map[string]interface{}{c}

